### PR TITLE
Program Homepage- Admin side & Mentor/Mentee/Unauthenticated side

### PR DESCRIPTION
**Pull request by innopoop**
Issue #25 

Routes Affected:
localhost:3000/programs/[slug]

Parts to review (sorry there are a lot & I probably should've separated text/image but the edits to files ended up overlapping with each other I apologize):
1. Editor Context- creates EditorState and initalizes plugins; takes care of disabled publish button state and conversion to and from JSON for homepage strings
2. RichTextEditor.tsx- handles key press (i.e. Ctrl + B => Bold) and Placeholder styling
3. TextStyles- Where all of the styles are defined and matched (For ease of altering in the future)
4. ToolBar- Toggle Style & Block Type buttons, Image Upload Button and verification (forgot to add alt tag for image); Must be nest within EditorProvider
5. Custom Image Plugin derived from [here](https://github.com/draft-js-plugins/draft-js-plugins/tree/f4b5e696ed1c8354d98ac53bc58372486589cf83/packages/image)- For alt handling and custom resized image wrapper component (addImage & index.tsx are pretty much the same as the github; Image.tsx has been changed to include resize wrapper)
6. ResizeableComponent- Resize Squares (UI), Resizing border
7. Publish Button
8. ReadOnlyHomepage

TODOS:
- [x] Fix bug related to initial getBoundingClientRect() returning a height and width of 0 => Resize squares not in the right position until you reclick
- [x] Vertical scaling of images
- [ ] Adding links to text
- [x] Determining how to save information about the homepage to database
- [x] Bug related to missing </div> tag in the RichTextEditor
- [ ] Host images and retrieve URLs

### CODE REVIEW

**Comment by kevinl120** [comment]
Two non-code things:
- Remove `.vscode/` or add it to `.gitignore`
- Remove `package.json` since we're using yarn.

**Comment by nova-admin** [comment]
#### 📝Changed routes:


| [**/create**](https://risc-eubv8suth-novaforgood.vercel.app/create) | [**/program/\[slug\]**](https://risc-eubv8suth-novaforgood.vercel.app/program/[slug]) |
|:-:|:-:|
|<a href="https://risc-eubv8suth-novaforgood.vercel.app/create">Screenshot of /create</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/8RlwlxMVNa45QbwPWQYeXwKJtOkHP6KmG9IEb8bTO5Dssnk_1aKIa216MIpap5NmPZG_tuCY6TAfjUS6lb3NLg.png">(view full size)</a> |<a href="https://risc-eubv8suth-novaforgood.vercel.app/program/[slug]">Screenshot of /program/[slug]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/d4aQ7SgLwm0mCqZDelugkNTS6BnlpY8V47sLseqOYjQ2yTwKoeS0vc6o8Vxvr5Qr1tKJMsgIKTismmJQzBO1oA.png">(view full size)</a> |







Commit 29e229936634c4c6ac279fb5d42752a2adef6d96 (https://risc-eubv8suth-novaforgood.vercel.app).

File: package.json [hunk 1]
```diff
@@ -14,8 +14,12 @@
     "@headlessui/react": "^1.2.0",
     "@svgr/webpack": "^5.5.0",
     "apollo-upload-client": "^15.0.0",
+    "@draft-js-plugins/editor": "^4.1.0",
+    "@draft-js-plugins/image": "^4.1.1",
```
Comment (innopoop) [inline package.json:18]: This is actually unnecessary

**Comment by innopoop** [comment]
image

**Pull request closed without merging**